docs/src/proposals/big-transactions.md

```diff
@@ -0,0 +1,186 @@
+# Big Transactions
+
+## Problem
+
+Messages transmitted to Solana validators must not exceed the IPv6 MTU size to
+ensure fast and reliable network transmission of cluster info over UDP.
+Solana's networking stack uses a conservative MTU size of 1280 bytes which,
+after accounting for headers, leaves 1232 bytes for packet data like serialized
+transactions.
+
+Developers building applications on Solana must design their on-chain program
+interfaces within the above transaction size limit constraint. One common
+work-around is to store state temporarily on-chain and consume that state in
+later transactions. This is the approach used by the BPF loader program for
+deploying Solana programs.
+
+However, this workaround doesn't work well when developers compose many on-chain
+programs in a single atomic transaction. With more composition comes more
+account inputs, each of which takes up 32 bytes. There is currently no available
+workaround for increasing the number of accounts used in a single transaction
+since each transaction must list all accounts that it needs to properly lock
+accounts for parallel execution. Therefore the current cap is about 35 accounts
+after accounting for signatures and other transaction metadata.
+
+## Proposed Solution
+
+Introduce a new on-chain account indexing program which stores account address
+mappings and add a new transaction format which supports concise account
+references through the new on-chain account indexes.
+
+### Account Indexing Program
+
+Here we describe a contract-based solution to the problem, whereby a protocol
+developer or end-user can create collections of related accounts on-chain for
+concise use in a transaction's account inputs. This approach is similar to page
+tables used in operating systems to succinctly map virtual addresses to physical
+memory.
+
+After addresses are stored on-chain in an index account, they may be succinctly
+referenced from a transaction using an index rather than a full 32 byte address.
+This will require a new transaction format to make use of these succinct indexes
+as well as runtime handling for looking up and loading accounts from the
+on-chain indexes.
+
```

Comment:
how about writing down bpf-visible changes?

- basically there should be none to maintain compatibility any given program's instruction is in normal tx or this new tx format
- but change is visible in the instruction sysvar?

Reply:
also what about rpc, explorer, cli? should we probably abstract away this and present as if the account keys are just large at the rpc and ui? CC: @CriesofCarrots @oJshua

Reply:
@ryoqun yes, no bpf visible changes should happen. The instruction sysvar uses decompiled instructions which will not know anything about account indexes.

As for client side impact. I think that RPC should support returning both compressed and decompressed transactions. I wrote up details here: https://github.com/solana-labs/solana/pull/17103/files#diff-e15f9fcbfd6f181fc2a36cf56d0b5bc1d6837fbd1da1d88a5739d12f6878d549R116-R120